cli/analytics.go

```diff
@@ -3,78 +3,121 @@ package cli
 import (
 	"encoding/json"
 	"fmt"
+	"io"
 	"os"
 
 	"github.com/bitrise-io/bitrise-plugins-analytics/analytics"
-	"github.com/bitrise-io/bitrise-plugins-analytics/configs"
 	"github.com/bitrise-io/bitrise/models"
 	"github.com/bitrise-io/bitrise/plugins"
 	log "github.com/bitrise-io/go-utils/log"
-	"github.com/hashicorp/go-version"
 	"github.com/pkg/errors"
 )
 
-// minBitriseCLIVersion points to the version of Bitrise CLI introduceses Bitrise plugins.
-const minBitriseCLIVersion = "1.6.0"
+// PayloadSource ...
+type PayloadSource interface {
+	Payload() (models.BuildRunResultsModel, error)
+}
 
-func ensureFormatVersion(pluginFormatVersionStr, hostBitriseFormatVersionStr string) (string, error) {
-	if hostBitriseFormatVersionStr == "" {
-		return fmt.Sprintf("This analytics plugin version would need bitrise-cli version >= %s to submit analytics", minBitriseCLIVersion), nil
-	}
+// SourceType ...
+type SourceType int
 
-	hostBitriseFormatVersion, err := version.NewVersion(hostBitriseFormatVersionStr)
-	if err != nil {
-		return "", errors.Wrapf(err, "failed to parse bitrise format version (%s)", hostBitriseFormatVersionStr)
+// SourceTypes ...
+const (
+	StdinSource SourceType = iota
+	EnvSource
+)
+
+// PayloadSourceFactory ....
+func PayloadSourceFactory(t SourceType) PayloadSource {
+	if t == StdinSource {
+		return StdinPayloadSource{os.Stdin}
 	}
+	return EnvPayloadSource{os.Getenv(plugins.PluginInputPayloadKey)}
+}
 
-	pluginFormatVersion, err := version.NewVersion(pluginFormatVersionStr)
+// StdinPayloadSource ....
+type StdinPayloadSource struct {
+	reader io.Reader
```

Comment:
if you have a single field, is it needed to have a struct type def or you can have a `type StdinPayloadSource io.Reader` also?